tests/compile-fail/environ-gets-deallocated.rs

```diff
@@ -0,0 +1,24 @@
+//ignore-windows: TODO env var emulation stubbed out on Windows
+
+#[cfg(target_os="linux")]
+fn get_environ() -> *const *const u8 {
+  extern "C" {
+    static environ: *const *const u8;
+  }
+  environ
```

Comment:
> use of extern static is unsafe and requires unsafe function or block

Reply:
Uhh it seems i focused on testing macos and forgot about linux

Reply:
It's hard to always test all platforms, that's why we have CI. :)

But with https://github.com/rust-lang/miri/issues/1198 you'll soon be able to test Linux, Windows and macOS targets on any host platform. :D

Reply:
that will be awesome, I've been waiting for that non-existent windows user that wants to implement all the shims we have now for the other two platforms.